src/client/mod.rs

```diff
@@ -158,20 +158,40 @@ impl Client {
         Database::new(self.clone(), name, Some(options))
     }
 
-    /// Gets information about each database present in the cluster the Client is connected to.
-    pub async fn list_databases(
+    async fn list_databases_common(
         &self,
         filter: impl Into<Option<Document>>,
         options: impl Into<Option<ListDatabasesOptions>>,
+        session: Option<&mut ClientSession>,
     ) -> Result<Vec<DatabaseSpecification>> {
         let op = ListDatabases::new(filter.into(), false, options.into());
-        self.execute_operation(op, None).await.and_then(|dbs| {
+        self.execute_operation(op, session).await.and_then(|dbs| {
             dbs.into_iter()
                 .map(|db_spec| bson::from_document(db_spec).map_err(crate::error::Error::from))
                 .collect()
         })
     }
 
+    /// Gets information about each database present in the cluster the Client is connected to.
+    pub async fn list_databases(
+        &self,
+        filter: impl Into<Option<Document>>,
+        options: impl Into<Option<ListDatabasesOptions>>,
+    ) -> Result<Vec<DatabaseSpecification>> {
+        self.list_databases_common(filter, options, None).await
+    }
+
+    /// Gets information about each database present in the cluster the Client is connected to.
```

Comment:
Can we add "using the provided `ClientSession`" to this doc (in the same vein as e.g. [this](https://docs.rs/mongodb/2.0.0-beta.2/mongodb/struct.Collection.html#method.aggregate_with_session))?

Reply:
Good call, done.